node/modules/storageminer.go

```diff
@@ -468,6 +469,13 @@ func BasicDealFilter(user dtypes.DealFilter) func(onlineOk dtypes.ConsiderOnline
 				return false, fmt.Sprintf("cannot seal a sector before %s", deal.Proposal.StartEpoch), nil
 			}
 
+			// Reject if it's more than 7 days in the future
+			// TODO: read from cfg
+			maxStartEpoch := ht + abi.ChainEpoch(7*builtin.EpochsInDay)
```

Comment:
Consider using `earliest` here.

Reply:
@schomatis this is actually the _latest_ epoch that the deal can start at, to prevent them for setting a start date really far in the future.  Sorry if the terminology was confusing.

Reply:
@ingar the suggestion is to use earliest in place of `ht` here -- cause if sealDuration > 7 days, then there is no acceptable window for StartEpoch.

We need to use
```
maxStartEpoch := earliest + abi.ChainEpoch(7*builtin.EpochsInDay)
```

Reply:
Got it.

Reply:
Follow-up fix in https://github.com/filecoin-project/lotus/pull/4337.